src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -89,13 +94,21 @@ public void storeMessage(Message message) {
     for (Entity entity : results.asIterable()) {
       try {
         String idString = entity.getKey().getName();
+        System.out.println("successfully read the message entity ID");
         UUID id = UUID.fromString(idString);
         String user = (String) entity.getProperty("user");
+        System.out.println("successfully read the message entity USER");
         String text = (String) entity.getProperty("text");
+        System.out.println("successfully read the message entity TEXT");
         long timestamp = (long) entity.getProperty("timestamp");
+        System.out.println("successfully read the message entity TIMESTAMP");
+        Float sentimentScore = (Float) entity.getProperty("sentimentScore");
```

Comment:
The error seems to be in this line. This is the message I see "[INFO] java.lang.ClassCastException: java.lang.Double cannot be cast to java.lang.Float". Does anyone know a fix around this?

Reply:
I think it means that entity.getProperty("sentimentScore") returns a value of type double and you can't just cast it to float with a (float). Is it a must to use type Float? Is it possible to just use double instead?

Reply:
Its very odd that its returning float value when you instantiated it as float below...hmm